### PR TITLE
Add licensed interaction via Actions.

### PR DESCRIPTION
**Pull request by patrickt**

### CODE REVIEW

**Comment by tclem** [comment]
@patrickt this is close now - I ended up doing license checks as part of the main build so that the dependencies are actually in the cabal store for checking (no reason to double build). There are a bunch of licenses that aren't formatted well and don't classify with `licensed`, which is a bummer and required some tedious manual checking, but we shouldn't have to do that too often.

**Comment by tclem** [comment]
Finally ready to go here. I ran into a number of build issues: the actions based licensed-ci seems to require the push event, not pull_request, and for some reason ci doesn't re-run when the licensed-ci bot pushes, but we do now have proper auditing of our open source licenses.

**Comment by patrickt** [comment]
Awesome - thank you so much, @tclem! I’m signing off on this, but would appreciate a pair of fresh eyes.

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -1,12 +1,6 @@
 name: Haskell CI
 
-on:
-  # Trigger the workflow on push or pull request,
-  # but only for the master branch
-  push:
-    branches:
-      - master
-  pull_request:
+on: [push]
```
Comment (patrickt) [inline .github/workflows/haskell.yml]: Won’t this turn CI off on PR builds?

**Comment by patrickt** [comment]
A thought: would it be easier to do this in a separate .yml file? Does that work?

**Comment by tclem** [comment]
> A thought: would it be easier to do this in a separate .yml file? Does that work?

I tried that first. The problem is that it appears that you need to build the entire project before you can run the licensing tool so then you've got two duplicate builds... I think this is unique to haskell/cabal as some of our other projects just read from a .lock file to enumerate dependencies and their licenses (and it's possible to have a separate workflow run).

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (tclem) [reply .github/workflows/haskell.yml]: No, but it will run CI on every push... which might not be ideal.

**Comment by robrix** [comment]
> > A thought: would it be easier to do this in a separate .yml file? Does that work?
> 
> I tried that first. The problem is that it appears that you need to build the entire project before you can run the licensing tool so then you've got two duplicate builds... I think this is unique to haskell/cabal as some of our other projects just read from a .lock file to enumerate dependencies and their licenses (and it's possible to have a separate workflow run).

In `semanticd` we could use a freeze file for that, but I’m not sure that’d work out for the library since we don’t want to freeze deps.

**Comment by patrickt** [comment]
This is moribund; returning to it later.

**Pull request closed without merging**